Source/creature_internal.cpp

```diff
@@ -79,13 +80,19 @@ TOperator* creature_internal::_get_genetic_operator(TMetadata** operator_metadat
 }
 
 crossover* creature_internal::_get_crossover(crossover_metadata** metadata, size_t num_crossovers) {
-    return _get_genetic_operator<crossover, crossover_metadata>(metadata, num_crossovers,
-            create_genetic_operator_a<crossover, crossover_metadata>(genetic_factory::create_crossover));
+    genetic_factory& factory = genetic_factory::get_instance();
+    create_genetic_operator_a<crossover, crossover_metadata>  create_crossover = [&factory](crossover_metadata* metadatai) {
+        return factory.create_crossover(metadatai);
```

Comment:
Is this a typo should it be `metadata`? Or did you mean to leave the `i` at the end?

Reply:
metadata is of type crossover_metadata** and is an 'array' of metadata for all the crossovers belonging to a particular chromosome. The create_crossover lambda expression will be working against a particular item (crossover_metadata*) in the metadata array and so I named the argument to create_crossover matadatai but perhaps its a stupid name.